instrumentation/google.golang.org/grpc/otelgrpc/filters.go

```diff
@@ -0,0 +1,204 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelgrpc // import "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+
+import (
+	"path"
+	"strings"
+)
+
+// Filter is a predicate used to determine whether a given request in
+// interceptor info should be traced. A Filter must return true if
+// the request should be traced.
+type Filter func(*interceptorInfo) bool
+
+type gRPCPath struct {
+	service string
+	method  string
+}
+
+// splitFullMethod splits path defined in gRPC protocol
+// and returns as gRPCPath object that has divided service and method names
+// https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
+// If name is not FullMethod, returned gRPCPath has empty service field.
+func splitFullMethod(name string) gRPCPath {
+	s, m := path.Split(name)
+	if s != "" {
+		s = path.Clean(s)
+		s = strings.TrimLeft(s, "/")
+	}
+	return gRPCPath{
+		service: s,
+		method:  m,
+	}
+}
+
+// Any takes a list of Filters and returns a Filter that
+// returns true if any Filter in the list returns true.
+func Any(fs ...Filter) Filter {
+	return func(i *interceptorInfo) bool {
+		for _, f := range fs {
+			if f(i) {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+// All takes a list of Filters and returns a Filter that
+// returns true only if all Filters in the list return true.
+func All(fs ...Filter) Filter {
+	return func(i *interceptorInfo) bool {
+		for _, f := range fs {
+			if !f(i) {
+				return false
+			}
+		}
+		return true
+	}
+}
+
+// None takes a list of Filters and returns a Filter that returns
+// true only if none of the Filters in the list return true.
+func None(fs ...Filter) Filter {
+	return func(i *interceptorInfo) bool {
+		for _, f := range fs {
+			if f(i) {
+				return false
+			}
+		}
+		return true
+	}
```

Comment:
```suggestion
	return Not(Any(fs...))
```

Isn't this the same?  If so, do we need the additional interface surface?